### PR TITLE
BaSP-TG-118: Employee Projects Page

### PR DESCRIPTION
**Pull request by fede230298**
Add new endpoint to get all projects by employee ID inside the project

### CODE REVIEW

File: src/controllers/projects.js [hunk 1]
```diff
@@ -63,6 +65,46 @@ const getProjectById = async (req, res) => {
...
+const getProjectsByEmployee = async (req, res) => {
+  try {
+    const { id } = req.params;
+    if (!isValidObjectId(id)) {
+      return res.status(400).json({
+        message: `Invalid id: ${id}`,
+        error: true,
+      });
+    }
+    const projects = await Projects.find().populate({
```
Comment (Ignacio-Sormani) [inline src/controllers/projects.js:77]: Can you just pass the id as a find argument and delegate the find logic?

**Comment by Ignacio-Sormani** [comment]
Tests and postman collection update is missing

File: src/controllers/projects.js [hunk 2]
```diff
@@ -63,6 +65,46 @@ const getProjectById = async (req, res) => {
...
+        path:
+      'employee',
+      },
+    });
+
+    const filteredProjects = projects.filter((project) => project.employees[0].employee.id === id);
+
+    if (!filteredProjects.length) {
+      return res.status(404).json({
+        message: `Employee with id ${id} not found`,
```
[the remark this answers is not in the document]
Comment (fede230298) [reply src/controllers/projects.js:89]: Done

[the remark this answers is not in the document]
Comment (MaicolFreire001) [reply src/controllers/projects.js]: I think this response shouldn't be an error, it just should return the empty array and the message should be managed in the FE at the moment to load the table

[the remark this answers is not in the document]
Comment (fede230298) [reply src/controllers/projects.js]: Yes, it was removed with the latest commit

File: src/controllers/projects.js [hunk 3]
```diff
@@ -63,6 +65,37 @@ const getProjectById = async (req, res) => {
...
+        message: `Invalid id: ${id}`,
+        error: true,
+      });
+    }
+    const projects = await Projects.find({ 'employees.employee': id })
+      .populate({
+        path: 'employees',
+        populate: {
+          path:
+      'employee',
```
Comment (Ignacio-Sormani) [inline src/controllers/projects.js:82]: This two lines should be one line: `path: 'employee',`

Comment (Ignacio-Sormani) [reply src/controllers/projects.js:82]: Fix it above too pls

File: src/controllers/projects.js [hunk 4]
```diff
@@ -63,6 +65,37 @@ const getProjectById = async (req, res) => {
...
+const getProjectsByEmployee = async (req, res) => {
+  try {
+    const { id } = req.params;
+    if (!isValidObjectId(id)) {
+      return res.status(400).json({
+        message: `Invalid id: ${id}`,
+        error: true,
+      });
+    }
+    const projects = await Projects.find({ 'employees.employee': id })
```
Comment (Ignacio-Sormani) [inline src/controllers/projects.js:77]: I have just one question. If an employee is assigned to more than one project, will it return an array of all the projects?

Comment (Ignacio-Sormani) [reply src/controllers/projects.js:77]: If the answer is yes, resolve the conversation

Comment (fede230298) [reply src/controllers/projects.js:77]: Yes

File: src/controllers/projects.js (hunk 3 above)
Comment (fede230298) [reply src/controllers/projects.js]: Done!

**Review by Ignacio-Sormani — APPROVED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

**Pull request merged**